iceoryx_utils/include/iceoryx_utils/internal/cxx/string.inl

```diff
@@ -343,6 +407,192 @@ inline bool operator!=(const char* const, const string<Capacity>&)
                   "before comparing it to a fixed string.");
     return false;
 }
+
+template <uint64_t Capacity>
+template <typename T>
+inline string<Capacity>& string<Capacity>::operator+=(const T&) noexcept
+{
+    static_assert(always_false<Capacity>::value,
+                  "operator += is not supported by cxx::string, please use append or unsafe_append instead");
```

Comment:
Seems like `operator+=` is supported https://en.cppreference.com/w/cpp/string/basic_string/operator%2B%3D but maybe not with C++14 ;-)